private/bufpkg/bufimage/util.go

```diff
@@ -118,11 +203,48 @@ func imageWithOnlyPaths(image Image, fileOrDirPaths []string, allowNotExist bool
 				return nil, fmt.Errorf("path %q has no matching file in the image", potentialDirPath)
 			}
 		}
+		for excludeFileOrDirPath := range excludeFileOrDirPathMap {
+			if _, ok := matchingPotentialExcludePathMap[excludeFileOrDirPath]; !ok {
+				// no match, this is an error given that allowNotExist is false
+				return nil, fmt.Errorf("path %q has no matching file in the image", excludeFileOrDirPath)
+			}
+		}
 	}
 	// we finally have all files that match fileOrDirPath that we can find, make the image
 	return getImageWithImports(image, nonImportPaths, nonImportImageFiles)
 }
 
+func excludeFile(
```

Comment:
Please include a comment that describes the algorithm from a high-level. We might also want to name this `excludeFiles` since it's excluding more than one file.